README.md

```diff
@@ -9,224 +9,429 @@ and understand how to run and work with Dgraph.
 ```
 
->WARNING: Be aware that there may be other .NET packages with similar names. To verify the official package, please visit https://www.nuget.org/packages/Dgraph. Make sure you are using the correct and official package to avoid potential confusion.
-
+> WARNING: Be aware that there may be other .NET packages with similar names. To verify
+> the official package, please visit https://www.nuget.org/packages/Dgraph. Make sure you
+> are using the correct and official package to avoid potential confusion.
 
 ## Supported Versions
 
-Each release of this client will support the equivalent Dgraph release. For example, 2020.03.XX will support any Dgraph instances with version 2020.03.XX. 
+Depending on the version of Dgraph that you are connecting to, you will have to use a
+different version of this client.
 
+| Dgraph version | Dgraph.net version  |
+| -------------- | ------------------- |
+| dgraph 21.X.Y  | Dgraph.net 21.3.1.2 |
+| dgraph 23.X.Y  | Dgraph.net 23.0.0   |
 
 ## Using a Client
 
+Most client and transaction methods return a `Result` object from the
+[FluentResults](https://github.com/altmann/FluentResults) library. You can check the status
+of the result by using `Result.IsSuccess` or `Result.IsFailed`. Exceptions are contained
+inside `Result.Errors`.
+
 ### Creating a Client
 
-Make a new client by passing in one or more GRPC channels pointing to alphas.
+An `IDgraphClient` can be created with a list of `GrpcChannel` as objects. Connecting
+to multiple Dgraph servers in the same cluster allows for better distribution of workload.
+
+The following code snippet shows just one connection.
+
+```c#
+using Dgraph;
+using Grpc.Net.Client;
+
+GrpcChannel channel = GrpcChannel.ForAddress("http://localhost:9080");
+using var dgraphClient = DgraphClient.Create(channel);
+```
+
+The connection can be configured by using `GrpcChannelOptions`.
 
 ```c#
-var uri = new Uri("http://127.0.0.1:9080");
+using Grpc.Net.Client;
+
 var options = new GrpcChannelOptions
 {
-    Credentials = ChannelCredentials.Insecure
+    CompressionProviders = <...>, // List of Grpc ICompressionProvider
+    Credentials = Grpc.Core.ChannelCredentials.Create(<...>)
 };
-var client = new DgraphClient(GrpcChannel.ForAddress(uri, options));
+GrpcChannel channel = GrpcChannel.ForAddress("http://localhost:9080", options);
 ```
 
+### Login into a namespace
+
+If your server has Access Control Lists enabled (Dgraph v1.1 or above), the client must be
+logged in for accessing data. Use `Login` to obtain and remember access and refresh JWTs.
+
+```c#
+var response = await dgraphClient.Login("user", "password");
+if (response.IsFailed) {
+    // Handle errors
+}
+```
+
+All subsequent operations via the logged in client will send along the stored access token.
+
+If your server additionally has namespaces (Dgraph v21.03 or above), use `LoginIntoNamespace`.
+
+```c#
+var response = await dgraphClient.LoginIntoNamespace("user", "password", 0x10);
```

Comment:
maybe use "groot" instead of "user"

Reply:
Also maybe use int instead of uint.

Reply:
The types and names of these arguments are from the `Dgraph.Api.LoginRequest` object generated by the proto files:

```
new LoginRequest
  {
      Userid = user, // string
      Password = password, // string
      Namespace = ns // ulong
  }
```

Should I still go ahead and change this? We'll have to cast the `int` back to `ulong`

Reply:
It should be fine. See https://github.com/dgraph-io/dgraph-js#multi-tenancy
In general go converts int to uint internally. For example, in Ratel and docs we do not state any `0x*` we use integers.

Reply:
So change the example but leave the argument type alone? 

i.e. only: `var response = await dgraphClient.LoginIntoNamespace("user", "password", 123);`

Reply:
yep

Reply:
Ok, done.